nav2_behaviors/src/behavior_server.cpp

```diff
@@ -124,6 +151,74 @@ BehaviorServer::loadBehaviorPlugins()
   return true;
 }
 
+void BehaviorServer::configureBehaviorPlugins()
+{
+  auto node = shared_from_this();
+
+  for (size_t i = 0; i != behavior_ids_.size(); i++) {
+    behaviors_[i]->configure(
+      node,
+      behavior_ids_[i],
+      tf_,
+      local_collision_checker_,
+      global_collision_checker_);
+  }
+}
+
+void BehaviorServer::setupResourcesForBehaviorPlugins()
+{
+  std::vector<nav2_core::CostmapInfoType> behaviors_info;
+
+  std::string local_costmap_topic, global_costmap_topic;
+  std::string local_footprint_topic, global_footprint_topic;
+  std::string robot_base_frame;
+  double transform_tolerance;
+  get_parameter("local_costmap_topic", local_costmap_topic);
+  get_parameter("global_costmap_topic", global_costmap_topic);
+  get_parameter("local_footprint_topic", local_footprint_topic);
+  get_parameter("global_footprint_topic", global_footprint_topic);
+  get_parameter("transform_tolerance", transform_tolerance);
+  get_parameter("robot_base_frame", robot_base_frame);
+
+  for (const auto & behavior : behaviors_) {
```

Comment:
Suggestion: This obviously works too, but maybe easier to read would be something like 

```
bool need_local_costmap = false; need_global_costmap = false;
for (const auto & behavior : behaviors_) {
  if (behavior->getResourceInfo() == both) {// set both to true and break, no reason to continue}
  if (behavior->getResourceInfo() == local) {local = true}
  if (behavior->getResourceInfo() == global) {global = true}
}
```

Then you can just use the  bools in the statements to create the objects which might be easier to read the `if` statement :-)

Reply:
The multi-line multi-scope if statement is a bit much potentially